### PR TITLE
feat: ADR-0006 hamlet engine management

### PR DESCRIPTION
**Pull request by roleyfoley**
## Intent of Change

- New feature (non-breaking change which adds functionality)

## Description

ADR for an update to the hamlet engine component release process

## Motivation and Context

This is an attempt to standardised and simplify the management of hamlet software component releases

### CODE REVIEW

**Comment by roleyfoley** [comment]
Haven't set an outcome yet as wanted to schedule a chat about it

**Comment by ml019** [comment]
Do we include the idea of a `hamletenv` command in this ADR or is this ADR more about the package/storage mechanism required? There is mention of use of shims so it feels like a logical extension to suggest an existing pattern for how shims would be managed.

**Comment by roleyfoley** [comment]
I think that would probably be more of the implementation rather than the architecture. I could see this as just another command in the python executor rather than having to implement a new component or process.

**Review by ml019 — APPROVED** [review]

**Comment by ml019** [comment]
I'm good for this to go. @RossMurr4y @kshychko @ken9ross let's get a couple of approvers on this. Feels like for ADRs we should require a couple of approvers before merge.

File: README.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ The records below are accepted Design records which we follow as part of hamlet
 * [ADR-0003](adr/0003-provide-implementation-tracking-on-records.md) - Include Implementation Log as part of Decision Records
 * [ADR-0004](adr/0004-provide-error-codes-on-handled-exception.md) - [Provide Users an Error Code on Fatal Error]
 * [ADR-0005](adr/0005-convention-based-logging-classes.md) - [Convention-based Logging Classes]
+* [ADR-0006](adr/0006-hamlet-docker-layer-release-mgmt.md) - [short title of solved problem and solution]
```
Comment (RossMurr4y) [inline README.md]:
```suggestion
* [ADR-0006](adr/0006-hamlet-docker-layer-release-mgmt.md) - [Hamlet Distribution with Docker Layers]
```

**Comment by ml019** [comment]
@RossMurr4y with you to approve

**Review by RossMurr4y — APPROVED** [review]

**Pull request merged**